src/ray/gcs/gcs_autoscaler_state_manager.cc

```diff
@@ -608,32 +608,35 @@ void GcsAutoscalerStateManager::CancelInfeasibleRequests() const {
   for (const auto &node_infeasible_request_pair : per_node_infeasible_requests) {
     const auto &node_id = node_infeasible_request_pair.first;
     const auto &infeasible_shapes = node_infeasible_request_pair.second;
-    const auto raylet_client = raylet_client_pool_.GetByID(node_id);
-
-    if (raylet_client) {
-      std::string resource_shapes_str =
-          ray::VectorToString(infeasible_shapes, ray::DebugString<std::string, double>);
-
-      RAY_LOG(WARNING) << "Canceling infeasible requests on node " << node_id
-                       << " with infeasible_shapes=" << resource_shapes_str;
-
-      raylet_client->CancelLeasesWithResourceShapes(
-          infeasible_shapes,
-          [node_id](const Status &status,
-                    const rpc::CancelLeasesWithResourceShapesReply &) {
-            if (status.ok()) {
-              RAY_LOG(INFO) << "Infeasible tasks cancelled on node " << node_id;
-            } else {
-              // Autoscaler will eventually retry the infeasible task cancellation
-              RAY_LOG(WARNING) << "Failed to cancel infeasible requests on node "
-                               << node_id
-                               << ". RPC failed with status: " << status.ToString();
-            }
-          });
-    } else {
+    auto node = gcs_node_manager_.GetAliveNode(node_id);
+    if (!node.has_value()) {
```

Comment:
If I'm understanding correctly, we previously relied on the unavailable timeout here, but now we're explicitly checking that the node is still alive?

Reply:
is this strictly required? we would still have the unavailable callback called once without it, right?